nir/src/main/scala/scala/scalanative/nir/Types.scala

```diff
@@ -38,7 +38,7 @@ object Type {
   final object F64 extends F(64)
 
   final case class Array(ty: Type, n: Int)              extends Type
-  final case class Function(args: Seq[Type], ret: Type) extends Type
+  final case class Function(args: Seq[Arg], ret: Type)  extends Type
```

Comment:
Thinking about it some more: More generally `isExtern` is probably a special case of calling convention (with variants c and fast calling conventions), similarly to how `sret` and `byval` are variants of passing conventions. Having a simple sealed hiearchy for each seems to be a good idea.